### PR TITLE
[SPARK-12537] [SQL] Add option to accept quoting of all character backslash quoting mechanism

### PR DESCRIPTION
**Pull request by Cazen**
We can provides the option to choose JSON parser can be enabled to accept quoting of all character or not.

### CODE REVIEW

**Comment by Cazen** [comment]
For example, if JSON file that includes not listed by JSON backslash quoting specification, it returns corrupt_record
JSON File
{"name": "Cazen Lee", "price": "$10"}
{"name": "John Doe", "price": "\$20"}
{"name": "Tracy", "price": "$10"}
corrupt_record(returns null)
scala> df.show
+--------------------+---------+-----+
|     _corrupt_record|     name|price|
+--------------------+---------+-----+
|                null|Cazen Lee|  $10|
|{"name": "John Do...|     null| null|
|                null|    Tracy|  $10|
+--------------------+---------+-----+
And after apply this patch, we can enable allowBackslashEscapingAnyCharacter option like below
scala> val df = sqlContext.read.option("allowBackslashEscapingAnyCharacter", "true").json("/user/Cazen/test/test2.txt")
df: org.apache.spark.sql.DataFrame = [name: string, price: string]

scala> df.show
+---------+-----+
|     name|price|
+---------+-----+
|Cazen Lee|  $10|
| John Doe|  $20|
|    Tracy|  $10|
+---------+-----+
This issue similar to HIVE-11825, HIVE-12717.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonParsingOptionsSuite.scala [hunk 1]
```diff
@@ -111,4 +111,21 @@ class JsonParsingOptionsSuite extends QueryTest with SharedSQLContext {
...
+    assert(df.schema.head.name == "_corrupt_record")
+  }
+
+  test("allowBackslashEscapingAnyCharacter on") {
+    val str = """{"name": "Cazen Lee", "price": "\$10"}"""
+    val rdd = sqlContext.sparkContext.parallelize(Seq(str))
+    val df = sqlContext.read.option("allowBackslashEscapingAnyCharacter", "true").json(rdd)
+
+    assert(df.schema.head.name == "name")
+    assert(df.first().getString(0) == "Cazen Lee")
```
Comment (rxin) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonParsingOptionsSuite.scala]: should we also test the price field?

**Comment by rxin** [comment]
@Cazen how come you closed the pull request?

**Comment by Cazen** [comment]
Hi Xin Thank you for review

I've created PR(11496, this PR) but it doesn't connect with jira(SPARK-12537) so I've closed.

After that, I recreated PR(11497) but linked 11496 in the jira instead of 11497 automatically.

Should I reopen this PR and close new one(11497)?

I'm sorry about confusing.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonParsingOptionsSuite.scala (hunk 1 above)
Comment (Cazen) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonParsingOptionsSuite.scala]: You right. It needeed
I'll modify test code soon
Thz

**Comment by rxin** [comment]
It's fine to use https://github.com/apache/spark/pull/10497

Just update it there.

**Pull request closed without merging**